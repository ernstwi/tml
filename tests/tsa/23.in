default location
    invar { x < 5 }
default edge
    guard { x == 10 }
    action { a }
    reset { x }

init foo invar { y <= 10 }
foo -> bar
    action { b }
    reset { z, y }
