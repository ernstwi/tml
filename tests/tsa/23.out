{
    "actions": [
        "b"
    ],
    "clocks": [
        "x",
        "y",
        "z"
    ],
    "edges": [
        {
            "action": "b",
            "from": "foo",
            "guard": [
                {
                    "clock": "x",
                    "clocks_n": 1,
                    "cmp": "eq",
                    "n": 10
                }
            ],
            "reset": [
                "z",
                "y"
            ],
            "to": "bar"
        }
    ],
    "locations": [
        {
            "id": "bar",
            "initial": false,
            "invariant": [
                {
                    "clock": "x",
                    "cmp": "lt",
                    "n": 5
                }
            ]
        },
        {
            "id": "foo",
            "initial": true,
            "invariant": [
                {
                    "clock": "y",
                    "cmp": "le",
                    "n": 10
                }
            ]
        }
    ]
}
